makeflow/src/makeflow_hook.c

```diff
@@ -55,45 +72,93 @@ struct dag_file * makeflow_hook_add_output_file(struct dag *d, struct batch_task
 }
 
 
-int makeflow_hook_register(struct makeflow_hook *hook) {
+int makeflow_hook_register(struct makeflow_hook *hook, struct jx **args) {
 	assert(hook);
-	if (!makeflow_hooks) makeflow_hooks = list_create();
+	if (!makeflow_hooks)     makeflow_hooks     = list_create();
+	if (!makeflow_hook_args) makeflow_hook_args = list_create();
+	if (!makeflow_hook_self) makeflow_hook_self = list_create();
 
 	/* Add hook by default, if it doesn't exists in list of hooks. */
 	int rc = MAKEFLOW_HOOK_SUCCESS;
 	struct makeflow_hook *h = NULL;
-	debug(D_MAKEFLOW_HOOK, "Hook %s:trying to registered",hook->module_name?hook->module_name:"");
+	struct jx *h_args = NULL;
+	debug(D_MAKEFLOW_HOOK, "Hook %s:trying to register",hook->module_name?hook->module_name:"");
 
 	if(hook->register_hook){
-		rc = hook->register_hook(hook, makeflow_hooks);
-	} else {
-		list_first_item(makeflow_hooks);
-		while((h = list_next_item(makeflow_hooks))){
-			if(!strcmp(h->module_name, hook->module_name)){
+		rc = hook->register_hook(hook, makeflow_hooks, args);
+	}
+
+	if(!(hook->register_hook) || rc == MAKEFLOW_HOOK_SKIP) {
+		struct list_cursor *cur = list_cursor_create(makeflow_hooks);
+		struct list_cursor *acur = list_cursor_create(makeflow_hook_args);
+		// This should be a reverse traversal so that the most recent
```

Comment:
I wasn't clear about this comment. Is this a TODO, or a description of how things work now?

Reply:
Changing to a reverse traversal with a break when we find the value.